### DISCUSSION TITLE
run_tests "ERROR: Cannot open connected subprocess" for several tests

### DISCUSSION THREAD

**Discussion by astroJarHead** [iraf-community/iraf#246] [Q&A] [answered]
After installing v 2.17 the result of test/run_tests was:
```
Test summary:      15 failed
    111 passed
      2 skipped
      1 xfailed
```
All the error message contained text similar to the following except the sub-process called was different:
```
Output
======
cl> copy noao$artdata/numrecipes.x .
cl> copy noao$artdata/gammln.c .
cl> softools
cl> xc -I$iraf/include test-gasdev.x numrecipes.x gammln.c
cl> task $test_gasdev = test-gasdev.e
cl> test_gasdev
ERROR: Cannot open connected subprocess (test-gasdev.e)
     called as: `cl ()'
Error while reading login.cl file - may need to rebuild with mkiraf
Fatal startup error.  CL dies.
```
Attached is the output of run_tests
[out.run_tests.txt](https://github.com/iraf-community/iraf/files/10230445/out.run_tests.txt)

sysgen did seem to finish without problems prior to the tests running. At least 
that I could note with my limited experience. The output from sysgen completed with:
```
Updated 31 files in libpkg.a
xc -c -w -p noao x_quad.x
xc -Nz -p noao x_quad.o libpkg.a -lxtools -o xx_quad.e
move `xx_quad.e' to `noaobin$x_quad.e'
purge directory `noaobin$'
!date
Wed Dec 14 09:30:16 MST 2022
```

**Comment by olebole** [comment, marked as answer]
It would still be useful to attach the build log. 
The reason here is unclear; it seems that with the system you (sometimes) can't compile own tasks -- some tasks however seem to work.

I would recommend to try them out manually in the "ecl" shell; first create a file test-gasdev.x with the content from [here](https://github.com/iraf-community/iraf/blob/main/test/numerical-recipes.md#gasdev-normally-distributed-deviate-of-zero-mean-and-unit-var).  Then start ecl and run the commands as in the log (resp. the link above). The log shows (lines starting with `cl>` ) what you should input and what you should expect. The **xc** task you have shown above should produce an output file `test-gasdev.e`, which should have the executable flag set.

I am afraid that you need to do some debugging here yourself.

**Reply by astroJarHead** [reply]
Thank you, I will do as you suggest. 

Attached is the build.log file. Perhaps a more experienced eye will see something helpful? 

[build.log](https://github.com/iraf-community/iraf/files/10230933/build.log)

**Reply by olebole** [reply]
The build log look OK.

**Reply by astroJarHead** [reply]
👍

**Reply by astroJarHead** [reply]
The manual test using the "ecl" shell was successful as the newly defined task test_gasdev did run. There was a difference in numerical precision:

ecl> copy noao$artdata/numrecipes.x .
ecl> copy noao$artdata/gammln.c .
ecl> softools
      generic     lroff       mkapropos   mkpkg       mkttydata   rmfiles     wtar        xyacc       
      hdbexamine  memchk      mkhelpdb    mktags      rmbin       rtar        xc          

softools> xc -I$iraf/include test-gasdev.x numrecipes.x gammln.c
softools> task $test_gasdev = test-gasdev.e
softools> test_gasdev
Mean     = 0.0000601690205889
Variance = 1.0002424406825960
Skew     = 0.0001202033929215
Kurtosis = 3.0006296332838071

I notice that the prompt in the the output I recorded from run_test that began this discussion indicated the "cl" shell 
was in use. I did repeat the manual test with the "cl" shell and it ran successfully.

**Reply by olebole** [reply]
OK; so everything seems to work now. And you can start IRAF and do something with it. Right?

**Reply by astroJarHead** [reply]
Yes. Other than the oddity with the run_tests all seems well. I appreciate your assistance.
